File: app/src/java/java/com/vktiwari/pagingretrofitrxjava/model/People.java
package com.vktiwari.pagingretrofitrxjava.model;

import android.os.Parcel;
import android.os.Parcelable;

public class People implements Parcelable {
    private static int increment = 0;

    private long id;
    public String name;
    public String height;
    public String mass;
    public String created;

    public People(){
        id = ++increment;
    }

    public long getId() {
        return id;
    }

    protected People(Parcel in) {
        name = in.readString();
        height = in.readString();
        mass = in.readString();
        created = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeString(height);
        dest.writeString(mass);
        dest.writeString(created);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<People> CREATOR = new Creator<People>() {
        @Override
        public People createFromParcel(Parcel in) {
            return new People(in);
        }

        @Override
        public People[] newArray(int size) {
            return new People[size];
        }
    };

    @Override
    public boolean equals(Object obj) {
        if (obj == this)
            return true;

        People people = (People) obj;
        return people.id == this.id;
    }

    @Override
    public String toString() {
        return "People{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", height='" + height + '\'' +
                ", mass='" + mass + '\'' +
                ", created='" + created + '\'' +
                '}';
    }
}

File: app/src/kotlin/java/com/vktiwari/pagingretrofitrxjava/model/People.kt
package com.vktiwari.pagingretrofitrxjava.model;

import android.os.Parcel;
import android.os.Parcelable;

class People : Parcelable {

    var id: Long = 0
    var name: String
    var height: String
    var mass: String
    var created: String

    init {
        id = (++increment).toLong()
    }

    constructor(`in`: Parcel) {
        name = `in`.readString()
        height = `in`.readString()
        mass = `in`.readString()
        created = `in`.readString()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeString(height)
        dest.writeString(mass)
        dest.writeString(created)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun equals(obj: Any?): Boolean {
        if (obj === this)
            return true

        val people = obj as People?
        return people!!.id == this.id
    }

    override fun toString(): String {
        return "People{" +
                "id=" + id +
                ", name='" + name + '\''.toString() +
                ", height='" + height + '\''.toString() +
                ", mass='" + mass + '\''.toString() +
                ", created='" + created + '\''.toString() +
                '}'.toString()
    }

    override fun hashCode(): Int {
        var result = increment
        result = 31 * result + id.hashCode()
        result = 31 * result + name.hashCode()
        result = 31 * result + height.hashCode()
        result = 31 * result + mass.hashCode()
        result = 31 * result + created.hashCode()
        return result
    }

    companion object {
        private var increment = 0
        @JvmField
        val CREATOR = object : Parcelable.Creator<People> {
            override fun createFromParcel(parcel: Parcel): People {
                return People(parcel)
            }

            override fun newArray(size: Int): Array<People?> {
                return arrayOfNulls(size)
            }
        }
    }
}
